src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -214,6 +216,39 @@ public function attributesToArray()
         return $attributes;
     }
 
+    /**
+     * Set attributes has append to the appends array.
+     *
+     * @return void
+     */
+    public function sutAppendMutatedAttributes()
+    {
+        $attributes = $this->getAppendMutatedAttributes();
+        $this->append($attributes);
+    }
+
+    /**
+     * Get attributes has append to the appends array.
+     *
+     * @return array
+     */
+    protected function getAppendMutatedAttributes()
+    {
+        $class = static::class;
+
+        return collect(static::getAttributeMarkedMutatorMethods($class))
+            ->filter(function ($methodName) {
+                $attribute = $this->{Str::camel($methodName)}();
+
+                return $attribute->withAppend;
```

Comment:
Since the result of `getAttributeMarkedMutatorMethods` is an array of the method names, can you not just do the following?

```suggestion
                return $this->{$methodName}()->withAppend;
```